src/libYARP_dev/src/yarp/dev/ReturnValue.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ * SPDX-FileCopyrightText: 2023-2023 Istituto Italiano di Tecnologia (IIT)
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <yarp/dev/ReturnValue.h>
+
+using namespace yarp::dev;
+
+yarp_ret_value::yarp_ret_value()
+{
+}
+
+yarp_ret_value::yarp_ret_value(const bool& val)
+{
+    if (val)
+    {
+        value_b = return_code::return_value_ok;
+    }
+    else
+    {
+        value_b = return_code::return_value_error_generic;
+    }
+}
+
+/*/yarp_ret_value::yarp_ret_value(const yarp_ret_value& other)
+{
+    this->value_b=other.value_b;
+}*/
+
+yarp_ret_value& yarp_ret_value::operator&=(const yarp_ret_value& other)
+{
+    if (other.operator bool() == true)
+    {
+        return *this;
+    }
+    else
+    {
+        value_b = return_code::return_value_error_generic;
+    }
+    return *this;
+}
+
+yarp_ret_value& yarp_ret_value::operator=(const bool& bool_val)
+{
+    if (bool_val)
+    {
+        value_b = return_code::return_value_ok;
+    }
+    else
+    {
+        value_b = return_code::return_value_error_generic;
+    }
+    return *this;
+}
+
+std::string yarp_ret_value::toString()
+{
+    switch (value_b)
+    {
+        case return_code::return_value_ok:
+            return std::string("ok");
+        case return_code::return_value_unitialized:
+            return std::string("return_value_unitialized");
```

Comment:
Typo: "uninitialized".

Suggestion: isn't the "return_value_" prefix a bit verbose? I'm also thinking of an overload to [LogStream](https://github.com/robotology/yarp/blob/c09c9f6d6e2f4c9c9fb3acf845459090c19652c0/src/libYARP_os/src/yarp/os/LogStream.h) so that this would be possible:

```cxx
double val;
auto ret = iMotor->getTemperature(0, &val);
yInfo() << ret; // prints "ok", "error_generic" "error_not_implemented" and so on
```

Reply:
That's a nice idea! Just give me a little more time before proceeding since we need to edit a few things more before.